tools/stats/upload_test_stats_running_jobs.py

```diff
@@ -0,0 +1,74 @@
+import sys
+import time
+from functools import lru_cache
+from typing import Any, List
+
+from tools.stats.test_dashboard import upload_additional_info
+from tools.stats.upload_stats_lib import get_s3_resource
+from tools.stats.upload_test_stats import get_tests
+
+
+BUCKET_PREFIX = "workflows_failing_pending_upload"
+
+
+@lru_cache(maxsize=None)
+def get_bucket() -> Any:
+    return get_s3_resource().Bucket("gha-artifacts")
+
+
+def delete_obj(key: str) -> None:
+    # Does not raise error if key does not exist
+    get_bucket().delete_objects(
+        Delete={
+            "Objects": [{"Key": key}],
+            "Quiet": True,
+        }
+    )
+
+
+def put_object(key: str) -> None:
+    get_bucket().put_object(
+        Key=key,
+        Body=b"",
+    )
+
+
+def do_upload(workflow_id: int) -> None:
+    workflow_attempt = 1
+    test_cases = get_tests(workflow_id, workflow_attempt)
+    # Flush stdout so that any errors in upload show up last in the logs.
+    sys.stdout.flush()
+    upload_additional_info(workflow_id, workflow_attempt, test_cases)
+
+
+def get_workflow_ids(pending: bool = False) -> List[int]:
```

Comment:
Maybe adding a comment about where these `WORKFLOW_ID.txt` comes from is useful here.  My understand is that `run_test` first puts the file on `workflows_failing_pending_upload`.  Then `read_s3` here picks it up, do the upload, then remove the file after that